### ISSUE TITLE
Update to Recent Files behavior for tracking open files

### ISSUE THREAD

**Issue by AlexSledge** [notepad-plus-plus/notepad-plus-plus#12330]
### Description of the Enhancement
The Recent Files behavior is in need of improvement.

### Expected Behavior
Option/Feature to have the Recent Files list be generated on file opens instead of (or in addition to) file save.

### User Stories
1) On application crash there would still be a record of the files which the user had open and was working on, and can easily re-open. User often leaves Notepad++ open for weeks with both multiple windows and multiple tabs within each of the windows.
2) If recording file save in addition to file open, it will be possible to filter the list of files down to the documents which remained open at the time of the application crash.
3) In the users I polled (7 co-workers) they all agreed their expectation was a history of files opened, with speculation that this seems to be the industry standard.

Note: In these samples, crash can also be interpreted as 'Windows decides to restart and close all their applications forcibly', and in instances where applicable can also simply mean the application is completely closed.

### Actual Behavior
No functionality for expected behavior.

### Debug Information
Notepad++ v8.4.4   (32-bit)

### Psychology
While people are more likely to complain about negative experiences, far fewer speak up when it is only an annoyance. This has been an annoyance to me for a long time, so I figured I'd make the request while I was here grabbing the latest. The few I've asked that use Notepad++ in any real capacity mentioned they experienced this 'quirk' themselves.

**Comment by dfs-** [comment]
Uh, Notepad++ is currently not adding files to the history on file open?
I noticed sometimes files are missing but never thought on that being the issue…

**Comment by alankilborn** [comment]
>On application crash

Why do you think the proposed data would be saved in this event?  If there's a crash then Notepad++ has no opportunity to save this data (with current scheme of this data being saved on exit).

"Design for crash" isn't a great plan.  :-(

------------------------------------------------

For me and I suspect a large number of others, there is no value in files that are already open into tabs appearing in the "recent files" area.  If I'm looking for something closed, I want the clutter (and that would be already-open files), well, not present.

I suppose if it is optional behavior, it is OK, but I think it was designed the way it is for definite reasons and doubt developers would see value in expending effort to change/augment it.

**Comment by AlexSledge** [comment]
This data is saved on file open, therefor it is already saved before any crash. If it crashes as you open a file, that file open may not be recorded, but as you just tried to open the file you should be able to determine which one it is.

**Comment by alankilborn** [comment]
>This data is saved on file open (in the ideal situation)

That would be ideal for this issue, but just to reiterate, it is NOT the current behavior.

**Comment by AlexSledge** [comment]
Yep, I agree 100% - not the current behavior. This is why I opened this feature request.

**Comment by pryrt** [comment]
The issue #12079 is very similar; the Community Forum posts linked in there might be of interest in this discussion as well.

**Comment by pryrt** [comment]
And to reply to @dfs- comment from a few days ago, which was somewhat skipped over:
> Uh, Notepad++ is currently not adding files to the history on file open? I noticed sometimes files are missing but never thought on that being the issue…

Nope, it adds files to the Recently Closed Files list on close.  That's why the [recent files history](https://npp-user-manual.org/docs/preferences/#recent-files-history) section of the manual includes the following:
![](https://community.notepad-plus-plus.org/assets/uploads/files/1650912171353-ac5a5323-98ec-4915-9ef0-d70c6c252ea3-image.png)

**Comment by pryrt** [comment]
Sorry, I got a little muddled.  The one I linked to was about session files, not recent history files.  Sorry for the noise
